### PR TITLE
Always build rhel images and patched images

### PR DESCRIPTION
**Pull request by l0rd**
Rhel images should be build when we do releases because it's better to
use release vs nightlies in prod.

Building patched images in ci build so that as soon as a PR is merged we
can use the new patched images as well (not only the new registry
image).

I have also done some refactoring:
- removed duplicate code
- renamed the script files consistently

This PR should be merged only when the cico jobs have been adapted: [PR](https://github.com/openshiftio/openshiftio-cico-jobs/pull/1049)

### CODE REVIEW

File: cico_functions.sh [hunk 1]
```diff
@@ -46,17 +46,36 @@ function install_deps() {
...
+
+function set_ci_tag() {
+  # Let's obtain the tag based on the 
+  # version defined in the 'VERSION' file
+  TAG=$(echo "$GIT_COMMIT" | cut -c1-"${DEVSHIFT_TAG_LEN}")
+  export TAG
+}
+
+function set_nightly_tag() {
+  # Let's obtain the tag based on the
```
Comment (ibuziuk) [inline cico_functions.sh]: @l0rd the comment looks incorrect

File: cico_functions.sh [hunk 2]
```diff
@@ -46,17 +46,36 @@ function install_deps() {
...
 
+function set_release_tag() {
+  # Let's obtain the tag based on the 
+  # version defined in the 'VERSION' file
+  TAG=$(head -n 1 VERSION)
+  export TAG
+}
+
+function set_ci_tag() {
+  # Let's obtain the tag based on the
```
Comment (ibuziuk) [inline cico_functions.sh]: @l0rd the comment looks incorrect

File: cico_functions.sh (hunk 1 above)
Comment (l0rd) [reply cico_functions.sh]: fixed

File: cico_functions.sh (hunk 2 above)
Comment (l0rd) [reply cico_functions.sh]: fixed

**Comment by ibuziuk** [comment]
@l0rd PR with cico jobs has been merged so, we need to update scripts names now - https://github.com/eclipse/che-devfile-registry/pull/64

**Comment by l0rd** [comment]
@ibuziuk please approve or I cannot merge otherwise

**Pull request merged**